com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceCreator.java

```diff
@@ -54,13 +57,23 @@ public void createService(CloudFoundryOperations client, CloudServiceExtended se
     }
 
     private List<String> computeValidServiceOfferings(CloudFoundryOperations client, List<String> possibleServiceOfferings,
-        List<String> existingServiceOfferings) {
+        String servicePlan, Map<String, List<CloudServicePlan>> existingServiceOfferings) {
         List<String> validServiceOfferings = new ArrayList<String>();
         for (String possibleServiceOffering : possibleServiceOfferings) {
-            if (existingServiceOfferings.contains(possibleServiceOffering)) {
+            for (String existingServiceOffering : existingServiceOfferings.keySet()) {
+                if (!existingServiceOffering.equals(possibleServiceOffering)) {
+                    LOGGER.warn(format("Service offering \"{0}\" is not existing", possibleServiceOffering));
+                    continue;
+                }
+                Optional<CloudServicePlan>  existingCloudServicePlan = existingServiceOfferings.get(possibleServiceOffering)
+                    .stream()
+                    .filter(cloudServicePlan -> servicePlan.equals(cloudServicePlan.getName()))
```

Comment:
cloudServicePlan and servicePlan are in my opinion too similar. Maybe servicePlan should be renamed to desiredServicePlan and cloudServicePlan should be just servicePlan?

Reply:
Done.